julia_src/reopt_model.jl

```diff
@@ -1207,34 +1215,45 @@ end
+		# SteamTurbine RE battery losses, RE curtailment, and exported RE terms are based on an approximation of percent RE because the general equation is nonlinear
+		# Thus, SteamTurbine %RE is only accurate if all techs that can supply ST have equal %RE fuel or provide equal quantities of steam to the steam turbine
 		SteamTurbinePercentREEstimate = @expression(m,
 			sum(p.TechPercentRE[tst] for tst in p.TechCanSupplySteamTurbine) / length(p.TechCanSupplySteamTurbine))
-		# Note: Steam turbine battery losses, curtailment, and exported RE terms are only accurate if all techs that can supply ST 
-		#		have equal RE%, otherwise it is an approximation because the general equation is non linear. 
-		SteamTurbineAnnualRETherm = @expression(m,p.TimeStepScaling *
+		AnnualSteamTurbineREThermOut = @expression(m,p.TimeStepScaling *
 			p.STThermOutToThermInRatio * sum(m[:dvThermalToSteamTurbine][tst,ts]*p.TechPercentRE[tst] for ts in p.TimeStep, tst in p.TechCanSupplySteamTurbine) # plus steam turbine RE generation 
-			- sum(m[:dvProductionToStorage][b,t,ts] * SteamTurbinePercentREEstimate * (1-p.ChargeEfficiency[t,b]*p.DischargeEfficiency[b]) for t in p.SteamTurbineTechs, b in p.HotTES, ts in p.TimeStep)) # minus battery storage losses from RE heat from steam turbine
+			- sum(m[:dvProductionToStorage][b,t,ts] * SteamTurbinePercentREEstimate * (1-p.ChargeEfficiency[t,b]*p.DischargeEfficiency[b]) for t in p.SteamTurbineTechs, b in p.HotTES, ts in p.TimeStep)) # minus battery storage losses from RE heat from steam turbine; note does not account for p.DecayRate
+		AnnualRESteamToSteamTurbine = @expression(m,p.TimeStepScaling *
+			sum(m[:dvThermalToSteamTurbine][tst,ts]*p.TechPercentRE[tst] for ts in p.TimeStep, tst in p.TechCanSupplySteamTurbine)) # steam to steam turbine from other techs- need to subtract this out from the total 	
+		AnnualSteamToSteamTurbine = @expression(m,p.TimeStepScaling *
+			sum(m[:dvThermalToSteamTurbine][tst,ts] for ts in p.TimeStep, tst in p.TechCanSupplySteamTurbine)) # steam to steam turbine from other techs- need to subtract this out from the total 	
 	end
 
 	# Renewable heat
-	m[:AnnualREHeatMMBTU] = @expression(m,p.TimeStepScaling*
+	m[:AnnualREHeatkWh] = @expression(m,p.TimeStepScaling*
 		(sum(p.ProductionFactor[t,ts] * p.LevelizationFactor[t] * m[:dvThermalProduction][t,ts] * p.TechPercentRE[t] for t in p.HeatingTechs, ts in p.TimeStep) #total RE heat generation (excl steam turbine, GHP)
 		- sum(m[:dvProductionToWaste][t,ts]* p.TechPercentRE[t] for t in p.CHPTechs, ts in p.TimeStep) #minus CHP waste heat
-		- sum(m[:dvProductionToStorage][b,t,ts]*p.TechPercentRE[t]*(1-p.ChargeEfficiency[t,b]*p.DischargeEfficiency[b]) for t in p.HeatingTechs, b in p.HotTES, ts in p.TimeStep) ) #minus thermal storage losses
-		+ SteamTurbineAnnualRETherm)
-	
+		+ sum(m[:dvSupplementaryThermalProduction][t,ts] * p.TechPercentRE[t] for t in p.CHPTechs, ts in p.TimeStep) # plus CHP supplemental firing thermal generation
```

Comment:
@Bill-Becker if you can also check the dvSupplementaryThermalProduction and SteamTurbine calcs that would be great. I still need to add in the cooling techs/loads.

Reply:
@kkrah looks good!